����finalproject300924.py

```diff
@@ -0,0 +1,157 @@
+def saveNewEntry():
+    global sum_of_ages, count_of_entries  # Allow global variables edit.
+    print("You selected Option 1 - Save a new entry")
+    name = input("Please enter a name: ")
+    # Check if ID is ok.
+    while True:
+        id_number = input("Please enter the ID of the name (numbers only): ")
+        if id_number.isdigit():
+            if id_number in entries:
+                print("ID " + id_number + " already exists.")
+                return  
+            else:
+                break  
+        else:
+            print("Invalid ID. Please enter numbers only.")
+    # Check if age is ok.
+    while True:
+        age = input("Please enter the age: ")
+        if age.isdigit():
+            age = int(age) 
+            break
+        else:
+            print("Invalid age. Please enter a valid number.")
+    # Save the new entry dic in another dic
+    entries[id_number] = {'name': name, 'age': age}
+    # Update sum of ages and count of entries
+    sum_of_ages += age
+    count_of_entries += 1
+    menu()

```

Comment:
תכתוב בבקשה את הקוד כך שאין קריאות לmenu בסוף כל פונקציה, זו לא הדרך.
צריכה להיות לך פונקציה מרכזית שקוראת לפונקציות לפי המספרים שלהן, ובסוף כל קריאה חוזרים אליה - והיא בלולאה. כך שאחרי שביצענו למשל את Save - חוזרים אחורה למי שקרא לסייב, יש press enter to continue, והלאה לאיטרציה הבאה - מציג את התפריט ושוב יכול לבחור אפשרות. לתקן בכל הקוד